homeassistant/components/binary_sensor/insteon_plm.py

```diff
@@ -56,32 +57,51 @@ def should_poll(self):
     @property
     def address(self):
         """Return the address of the node."""
-        return self._address
+        return self._insteon_device.address.human
 
     @property
-    def name(self):
-        """Return the name of the node."""
-        return self._name
+    def group(self):
+        """Return the INSTEON group that the entity responds to."""
+        return self._insteon_device_state.group
 
     @property
-    def is_on(self):
-        """Return the boolean response if the node is on."""
-        sensorstate = self._plm.get_device_attr(self._address, 'sensorstate')
-        _LOGGER.info("Sensor state for %s is %s", self._address, sensorstate)
-        return bool(sensorstate)
+    def name(self):
+        """Return the name of the node (used for Entity_ID)."""
+        name = ''
+        if self._insteon_device_state.group == 0x01:
+            name = self._insteon_device.id
+        else:
+            name = '{:s}_{:d}'.format(self._insteon_device.id,
+                                      self._insteon_device_state.group)
+        return name
 
     @property
     def device_state_attributes(self):
         """Provide attributes for display on device card."""
         insteon_plm = get_component('insteon_plm')
         return insteon_plm.common_attributes(self)
 
-    def get_attr(self, key):
-        """Return specified attribute for this device."""
-        return self._plm.get_device_attr(self.address, key)
-
     @callback
-    def async_binarysensor_update(self, message):
+    def async_binarysensor_update(self, deviceid, statename, val):
         """Receive notification from transport that new data exists."""
-        _LOGGER.info("Received update callback from PLM for %s", self._address)
-        self._hass.async_add_job(self.async_update_ha_state())
+        self.async_schedule_update_ha_state()
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._sensor_type
+
+    @property
+    def is_on(self):
+        """Return the boolean response if the node is on."""
+        sensorstate = self._insteon_device_state.value
+        return bool(sensorstate)
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register INSTEON update events."""
+        _LOGGER.debug('Device %s entity %s added. Now registering callback.',
+                      self.address, self._insteon_device_state.name)
+        self.hass.async_add_job(
+            self._insteon_device_state.register_updates,
```

Comment:
If this just appends the callback to a list, that is async safe, and then you do not need to use `async_add_job`.